packages/client/src/authentication/Authentication.ts

```diff
@@ -0,0 +1,43 @@
+import { invariant } from '@lens-protocol/shared-kernel';
+import { GraphQLClient } from 'graphql-request';
+
+import { getSdk, Sdk } from './graphql/auth.generated';
+
+export type Credentials = { accessToken: string; refreshToken: string };
+
+export class Authentication {
+  private readonly sdk: Sdk;
+
+  constructor(backendUrl: string) {
+    const client = new GraphQLClient(backendUrl);
+    this.sdk = getSdk(client);
+  }
+
+  async challenge(address: string): Promise<string> {
+    const result = await this.sdk.AuthChallenge({ address });
+
+    return result.data.result.text;
+  }
+
+  async verify(accessToken: string): Promise<boolean> {
+    const result = await this.sdk.AuthVerify({ accessToken });
+
+    return result.data.result;
+  }
+
+  async authenticate(address: string, signature: string): Promise<Credentials> {
+    const result = await this.sdk.AuthAuthenticate({ address, signature });
+
+    invariant(result.data, 'Not able to generate credentials. Credentials data not found.');
```

Comment:
I recently addressed this in the `@lens-protocol/react` SDK. That implementation if very ApolloClient specific, how to errors propagates from the response to the call site via `GraphQLClient` and the generated code?

Reply:
is it fine if I just remove these `invariant` checks for now, and proper error handling we will add a bit later?

Reply:
Sure, do you mind keep track of looking into cohesive approach to error handling with a dedicated ticket?

Reply:
ticket created https://avara.height.app/lens-sdk/T-5590